### PR TITLE
fix: Series Title Custom Charts

### LINKED ISSUE #16132: [Bug]: CustomChart | Unable to retrieve all info of the SelectedDataPoint
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Screenshot 2022-08-18 at 5 19 28 PM

### Steps To Reproduce

1. D&D Chart & select CustomChart
2. Give the below data in SeriesData:
```
{
	"type": "msstackedcolumn2d",
	"dataSource": {
		"chart": {
			"xaxisname": "Type",
			"yaxisname": "No. of Gyms",
			"theme": "fusion"
		},
		"categories": [
			{
				"category": [
					{
						"label": "Value"
					},
					{
						"label": "Elite"
					},
					{
						"label": "Pro"
					},
					{
						"label": "Lux"
					}
				]
			}
		],
		"dataset": [
			{
				"dataset": [
					{
						"seriesname": "Cult_gym",
						"data": [
							{
								"value": 18
							},
							{
								"value": 51
							},
							{
								"value": 29
							},
							{
								"value": 66
							}
						]
					},
					{
						"seriesname": "Cult_GX",
						"data": [
							{
								"value": 23
							},
							{
								"value": 42
							},
							{
								"value": 33
							},
							{
								"value": 64
							}
						]
					},
					{
						"seriesname": "Other_gym",
						"data": [
							{
								"value": 7
							},
							{
								"value": 15
							},
							{
								"value": 6
							},
							{
								"value": 14
							}
						]
					},
					{
						"seriesname": "Other_Gx",
						"data": [
							{
								"value": 4
							},
							{
								"value": 11
							},
							{
								"value": 10
							},
							{
								"value": 27
							}
						]
					}
				]
			}
		]
	}
}
```
3. Create a JS to read the 'seriesname' from the selected data point
4. **Observe: Unable to read the 'seriesname' value & there is a empty seriesTitle in the response**
5. **Expected : User should be able to read the selected datapoint values fine**

### Public Sample App

_No response_

### Version

Prod/Release v1.7.11

### PR DESCRIPTION
**Pull request by dhruvikn**
## Description

TLDR; This PR fixes the bug which was NOT sending the series title of the selected point in custom charts.

Previously, the series title was picked up from `dataSource.seriesName` which does not exist in the chart configs. 
The observation was that the series name exists individually for the datasets. So now, in the case of custom charts, we fetch the respective series name from the clicked data point.
 
Untitled-2022-07-21-1423

Fixes #16132

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manually tested

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by jsartisan — APPROVED** [review]

**Comment by chandannkumar** [comment]
Tested this PR and it is working as expected.
- Able to retrieve all the data including series Title from selectedDataPoint on custom chart
